interfaces/apparmor/spec_test.go

```diff
@@ -280,3 +280,75 @@ func (s *specSuite) TestApparmorSnippetsFromLayout(c *C) {
 	c.Assert(updateNS[3], Equals, profile3)
 	c.Assert(updateNS, DeepEquals, []string{profile0, profile1, profile2, profile3})
 }
+
+func (s *specSuite) TestChopTree(c *C) {
+	for _, tc := range []struct {
+		p    string   // path
+		d    int      // depth
+		l, r []string // left and right path expressions
+		e    string   // error pattern, if non-empty
+	}{
+		// Test case from the documentation of the function.
+		{p: "/foo/bar/froz/baz/", d: 3, // Assume first three directories exist
+			l: []string{"/", "/foo/", "/foo/bar/"},
+			// Assume that /foo/bar/froz and beyond may be missing
+			r: []string{"/foo/bar/*", "/foo/bar/*/", "/foo/bar/froz/*", "/foo/bar/froz/*/"}},
+
+		// Exhaustive test cases for directory paths.
+
+		{p: "/foo/bar/froz/", d: 0, // Assume that no directories exist
+			r: []string{"/*", "/*/", "/foo/*", "/foo/*/", "/foo/bar/*", "/foo/bar/*/"}},
```

Comment:
Below with files, you check for `l: []string(nil),` but you have no 'l' check here.

Reply:
l: []string(nil) is the default as it is a golang zero value, should just remove it elsewhere to be consistent